src/lib/y2network/clients/auto.rb

```diff
@@ -122,10 +122,12 @@ def merge_current_config?
       # Whatever is needed in /etc/hosts during AY installation
       def update_etc_hosts
         config = Yast::Lan.find_config(:yast)
-        return if !config
+        return if [nil, ""].include? config&.hostname&.static
 
-        static_connections = config.connections.select(&:static?)
+        static_connections = config.connections.select { |c| c.static? && c.ip&.address }
```

Comment:
in my POV it should be

```&& c.ip```

as ip is the place where nil can get into at https://github.com/yast/yast-network/blob/master/src/lib/y2network/autoinst/interfaces_reader.rb#L128 when no ip address is defined in profile.

btw ```ip``` and ```ip.address``` is non nil up to here as it gets non empty default here https://github.com/yast/yast-network/blob/master/src/lib/y2network/connection_config/base.rb#L91

Reply:
I just tried to avoid any other corner case. That is having **c.ip** without address.

Reply:
> I just tried to avoid any other corner case. That is having **c.ip** without address.

yes ... i understand this part, but main point should be



> in my POV it should be
> 
> `&& c.ip`
> 
> as ip is the place where nil can get into at https://github.com/yast/yast-network/blob/master/src/lib/y2network/autoinst/interfaces_reader.rb#L128 when no ip address is defined in profile.
> 
because

```config.ip = load_ipaddr(interface_section)```

and

```
def load_ipaddr(section)
     return if section.ipaddr.empty?
     <--- snip --->
```

Reply:
Take into account merged config, I can modify if you really want it, but will not make any difference at all, the current one is safer.

Reply:
Updated